MaterialLib/SolidModels/CreateLinearElasticIsotropic.h

```diff
@@ -16,16 +16,18 @@ namespace MaterialLib
 {
 namespace Solids
 {
-
 template <int DisplacementDim>
 std::unique_ptr<LinearElasticIsotropic<DisplacementDim>>
 createLinearElasticIsotropic(
     std::vector<std::unique_ptr<ProcessLib::ParameterBase>> const& parameters,
-    BaseLib::ConfigTree const& config)
+    BaseLib::ConfigTree const& config, const bool skip_type_checking)
 {
-    //! \ogs_file_param{material__solid__constitutive_relation__type}
-    config.checkConfigParameter("type", "LinearElasticIsotropic");
-    DBUG("Create LinearElasticIsotropic material");
+    if (!skip_type_checking)
```

Comment:
Yes, that is my purpose.

For this, I want to change the syntax of constitutive_relation input as
```
 <constitutive_relation>
       <elasticity>
        <youngs_modulus>E</youngs_modulus>
        <poissons_ratio>nu</poissons_ratio>
      </elasticity>
       <inelasticity>
            <type>CreepBGRa</type>
               ....
       </inelasticity>
 </constitutive_relation>
```

Reply:
With PR, the type check in createLinearElasticIsotropic is only omitted when CreepBGRa is used.

Reply:
I think this change is nice, considering we may want to combine different material models with different forms of elasticity tensors in the future (e.g. transversely isotropic or orthotropic)

Reply:
@nagelt: That sounds like "material model" and "form of elasticity tensors" are different concepts. So we should have an excessive discussion about how to represent them in the most correct, general and abstract way in C++ :rofl:

Reply:
I agree with @nagelt . Since Hook's laws is the base of the constitutive relationship, I think we can have a single XML element for elastic parameters. The input may also look like

```
<constitutive_relation>
    <type>CreepBGRa</type>
    <elasticity>
        <youngs_modulus>E</youngs_modulus>
        <poissons_ratio>nu</poissons_ratio>
    </elasticity>
     ....other parameters.
</constitutive_relation>
```